### ISSUE TITLE
Add high-dimensional manifolds

### ISSUE THREAD

**Issue by Filco306** [scikit-tda/tadasets#10]
Hello! 

I think high-dimensional manifolds, that is, manifolds beyond 2 and 3 dimensions would be very interesting to add. d-dimensional spheres have already been added, but if somebody knows how generate other types of manifolds of higher dimensions, I would be happy to cooperate and contribute to create this for this package :) 

Cheers,
Filip

**Comment by ctralie** [comment]
Yes, I would love to help with this.  Uniformly sampling manifolds is
tricky, but naively I would start by just doing furthest points sampling to
make sure things are even

**Comment by Filco306** [comment]
How great to hear! I think #11 is a start, but what other high-dimensional manifolds can we construct?

**Comment by Filco306** [comment]
Have a look [here](http://planning.cs.uiuc.edu/node137.html). He claims an n-dimensional Torus can simply be built by taking the cartesian of S^1, seen [here](http://planning.cs.uiuc.edu/node136.html). Can we use this perhaps?

**Comment by ctralie** [comment]
Yeah actually, a flat torus is very straightforward that way.  Simply use
meshgrid and have pairs of dimensions be a different circle, as you said

**Comment by Filco306** [comment]
Hmm, I am not sure I understand how you mean. Do you perhaps have some sample code?

**Comment by ctralie** [comment]
For 10,000 samples on a flat 2-torus, for example, one could do this

N = 100
t = np.linspace(0, 2*np.pi, N)
theta, phi = np.meshgrid(t, t)
theta = theta.flatten()
phi = phi.flatten()
X = np.zeros((N*N, 4))
X[:, 0] = np.cos(theta)
X[:, 1] = np.sin(theta)
X[:, 2] = np.cos(phi)
X[:, 3] = np.sin(phi)

**Comment by Filco306** [comment]
Very sorry for such a late reply. That is very interesting, I suppose we can extend that to an arbitrary number of dimensions?

**Comment by Filco306** [comment]
Would something like this do as a first draft? 

```python3
def sample_4d_torus(n_points, seed):
    assert np.sqrt(n_points) % 1 == 0, "Please pick a number of points with integer root. "
    np.random.seed(seed)
    N = int(np.sqrt())
    t = np.random.uniform(0,2*np.pi, N)
    t = np.linspace(0, 2*np.pi, N)
    theta, phi = np.meshgrid(t, t)
    theta = theta.flatten()
    phi = phi.flatten()
    X = np.zeros((N*N, 4))
    X[:, 0] = np.cos(theta)
    X[:, 1] = np.sin(theta)
    X[:, 2] = np.cos(phi)
    X[:, 3] = np.sin(phi)
    return X
```

**Comment by ctralie** [comment]
Yeah, for a flat torus in 4D, that should be uniform.  Did you mean to say
N = int(np.sqrt(n_points)) ?

**Comment by Filco306** [comment]
Oh yes, it should `int(np.sqrt(n_points))`. Can we extend this to more than 4 dimensions?

**Comment by ctralie** [comment]
Yup, you can do this sort of thing in every even dimension at 4 and beyond
